### PR TITLE
Update go-server template to include the conventional generated code line

### PR DESCRIPTION
**Pull request by viswajithiii**
Follow the convention from https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source.

The generated go client code already follows this: https://github.com/OpenAPITools/openapi-generator/blob/f357be480e987827fb4bdf1f0d1b4a4ad20f1f85/modules/openapi-generator/src/main/resources/go/partial_header.mustache#L18

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request. cc: @lwj5

### CODE REVIEW

**Comment by wing328** [comment]
please follow step 3 to update the samples

cc 
@antihax (2017/11) @grokify (2018/07) @kemokemo (2018/09) @jirikuncar (2021/01) @ph4r5h4d (2021/04) @lwj5 (2023/04)

**Comment by viswajithiii** [comment]
> please follow step 3 to update the samples
> 
> cc @antihax (2017/11) @grokify (2018/07) @kemokemo (2018/09) @jirikuncar (2021/01) @ph4r5h4d (2021/04) @lwj5 (2023/04)

Done!

**Comment by lwj5** [comment]
Good idea!

Although we could put it anywhere as long as it comes before any code, I typically see this comment place at the first line. This is easier for us as we can just read the first line and know it is generated.

It's up to you if you want to change it. 

LGTM as is.

**Comment by lwj5** [comment]
Both are within spec. Just that for us humans, we have to read all the other comments before coming to this one.

I'll leave it up to you

**Comment by viswajithiii** [comment]
> Both are within spec. Just that for us humans, we have to read all the other comments before coming to this one.
> 
> I'll leave it up to you

Sounds good, done! Let me know if you want me to do the same for the client code (I wasn't sure if we wanted to make a change in case we break some people who were assuming that it was the bottom line)

File: modules/openapi-generator/src/main/resources/go-server/partial_header.mustache [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// Code generated by OpenAPI Generator (https://openapi-generator.tech); DO NOT EDIT.
+
 /*
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/go-server/partial_header.mustache:3]: i'll file another PR to move it inside the /* .. */ comment block below to make the comment a bit more consistent.

Comment (wing328) [reply modules/openapi-generator/src/main/resources/go-server/partial_header.mustache:3]: filed https://github.com/OpenAPITools/openapi-generator/pull/18319

**Pull request merged**